_dev/apps/ui/src/store/modules/product-feed/actions.ts

```diff
@@ -1,54 +1,81 @@
+import appGetters from '@/store/modules/app/getters-types';
+import {type FullState, RequestState} from '@/store/types';
+import {formatMappingToApi} from '@/utils/AttributeMapping';
+import {deleteProductFeedDataFromLocalStorage, getDataFromLocalStorage} from '@/utils/LocalStorage';
 import {runIf} from '@/utils/Promise';
-import {ShippingSetupOption} from '@/enums/product-feed/shipping';
-import {fromApi, toApi} from '@/providers/shipping-rate-provider';
-import {
+import ActionsTypes from './actions-types';
+import MutationsTypes from './mutations-types';
+import type {
+  ProductFeedAPIPayload,
   ProductFeedSettings, ProductVerificationIssue, ProductVerificationIssueProduct, State,
 } from './state';
-import {formatMappingToApi} from '@/utils/AttributeMapping';
-import {IncrementalSyncContext} from '@/components/product-feed-page/dashboard/feed-configuration/feed-configuration';
-import {FullState, RequestState} from '@/store/types';
-import appGetters from '@/store/modules/app/getters-types';
-import {ProductIssue} from '@/components/render-issues/types';
-import ProductsStatusType from '@/enums/product-feed/products-status-type';
 
 type Context = ActionContext<State, FullState>;
 
 // ToDo: Get DTO type from API sources
-export const createProductFeedApiPayload = (settings:any) => ({
-  autoImportTaxSettings: settings.autoImportTaxSettings,
-  shippingSetup: settings.shippingSetup,
-  targetCountries: settings.targetCountries,
-  ...(
-    (settings.shippingSetup === ShippingSetupOption.ESTIMATE) ? {
-      rate: settings.rate,
-      estimateCarriers: settings.estimateCarriers,
-    } : {}
-  ),
-  ...(
-    (settings.shippingSetup === ShippingSetupOption.IMPORT) ? {
+export const createProductFeedApiPayload = (settings: ProductFeedAPIPayload): ProductFeedAPIPayload => {
+  // const productFeedApiPayload: Partial<ProductFeedAPIPayload>= {
+  const productFeedApiPayload: ProductFeedAPIPayload = {
+    autoImportTaxSettings: settings.autoImportTaxSettings,
+    shippingSetup: settings.shippingSetup,
+    targetCountries: settings.targetCountries,
+
+    // biome-ignore lint/suspicious/noExplicitAny: TYPE:MUST_FIX
+    attributeMapping: formatMappingToApi(settings.attributeMapping as any),
+    selectedProductCategories: settings.selectedProductCategories,
+    requestSynchronizationNow: settings.requestSynchronizationNow,
+  };
+  // TODO: refactor so it is Typescript valid
+  const filterShippingSettings = (s: ShopShippingInterface) => (
+    (s.collection !== ShopShippingCollectionType.CARRIERS || (!!s.properties.active && !s.properties.deleted))
+      && (!s.properties.country_ids || settings.targetCountries?.some((tc: string) => s.properties.country_ids.includes(tc))));
+
+  switch (settings.shippingSetup) {
+    case ShippingSetupOption.ESTIMATE:
+      productFeedApiPayload.rate = settings.rate;
+      productFeedApiPayload.estimateCarriers = settings.estimateCarriers?.map(filterEstimateCarriers);
```

Comment:
If we want to improve the types used in the product feed configuration, we could make this attribute mandatory when the Estimate is chosen (no undefined

Reply:
ok. but it seems to be set only if `settings.shippingSetup === ShippingSetupOption.ESTIMATE`
If I remove the optional flag on this field, that means, I'll have to always set it to a value, [].
Here :
```typescript
export const createProductFeedApiPayload = (settings: ProductFeedAPIPayload): ProductFeedAPIPayload => {
  const productFeedApiPayload: ProductFeedAPIPayload = {
    autoImportTaxSettings: settings.autoImportTaxSettings,
    shippingSetup: settings.shippingSetup,
    targetCountries: settings.targetCountries,

    // always need to set it
    estimateCarriers: [],

    attributeMapping: formatMappingToApi(settings.attributeMapping as any),
    selectedProductCategories: settings.selectedProductCategories,
    requestSynchronizationNow: settings.requestSynchronizationNow,
  };
```
if that's ok with you, I'll make the change.

Reply:
I think we should have two completely different types to be safe. One for the state in which some attributes are always define, one for the API payload in which you put some keys as optional.